pkg/mimir/mimir.go

```diff
@@ -626,6 +630,29 @@ func InheritCommonFlagValues(log log.Logger, fs *flag.FlagSet, common CommonConf
 				return fmt.Errorf("can't inherit common flags for %q: %w", desc, err)
 			}
 		}
+		if common.ClusterValidationLabel == "" {
+			// Nothing to inherit because origin was not set.
+			continue
+		}
+		for _, grpcClientConfig := range inheritance.ClusterValidationLabel {
+			if grpcClientConfig == nil {
+				continue
+			}
+			if grpcClientConfig.ClusterValidationLabel != "" {
+				// Can't inherit because destination was set.
+				continue
+			}
+			if common.ClusterValidationLabel == grpcClientConfig.ClusterValidationLabel {
+				// Already the same, no need to touch.
+				continue
+			}
+			level.Debug(log).Log(
+				"msg", "Inheriting flag value",
+				"flag_name", "cluster_value_label", "origin_value", common.ClusterValidationLabel,
+				"destination_value", grpcClientConfig.ClusterValidationLabel,
+			)
+			grpcClientConfig.ClusterValidationLabel = common.ClusterValidationLabel
+		}
```

Comment:
The fact that this can't reuse the existing inheritance code makes me a little bit sad, maybe we could reconsider the configuration on the dskit in a way that we could just reuse the common inheritance here?

Note that if we follow this path, it will become quite more complex once you design the http counterpart of this (which is IMO a reason to have it all in the same PR, if you don't want to rework it all again).